src/envoy/utils/utils.cc

```diff
@@ -118,6 +120,24 @@ bool GetPrincipal(const Network::Connection* connection, bool peer,
   return false;
 }
 
+bool GetSourceNamespace(const std::string& principal,
```

Comment:
please add comment on how the expected format

Reply:
Done.